src/mono/wasm/build/WasmApp.Native.targets

```diff
@@ -107,6 +107,9 @@
       <WasmBuildNative Condition="'$(RunAOTCompilation)' == 'true' and '$(RunAOTCompilationAfterBuild)' == 'true'">true</WasmBuildNative>
 
       <WasmBuildNative Condition="'$(WasmBuildNative)' == '' and @(NativeFileReference->Count()) > 0" >true</WasmBuildNative>
+      <!-- need to re-link dotnet.js when targeting legacy or CommonJS -->
```

Comment:
Isn't this that you need to re-link if *not* targeting legacy/commonJS?

Reply:
Fixed comment. This will be correct after we flip the default to ES6.